enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+
+### Upgrade Reconciliation
+
+```bash
+$ oc adm credentials reconcile --available-updates
+```
+
+This command will iterate all currently available updates in ClusterVersion.AvailableUpdates. For each given release image we will extract each CredentialsRequest and use the local users environment variables/credentials file to compare and adjust permissions as necessary.
+
+This command reconciles permissions *in an additive fashion*. We do not want to reconcile any reduction in permissions as that would affect the running components in the cluster right now. The command would print a warning if this situation is encountered, and it can be resolved after upgrade with the next command.
+
+For net new CredentialsRequests in the update, we have a problem. We cannot assume to store a secret in the target namespace as it may be for a component that does not exist in the currently running OpenShift release. To work around this problem we propose copying net new CredentialsRequests into the cluster (all CredentialsRequests live in the openshift-cloud-credential-operator namespace today), and introducing the concept of temporary secret storage. This would allow us to mint the credentials when we have the root creds available, grab the secrets which we typically only get at creation time and cannot retrieve later, and transfer them to the correct namespace when it is available.
+
+Each available update we process successfully is added to the reconciledAvailableUpdates list.
+
+### On-Going Reconciliation
+
+```bash
+$ oc adm credentials reconcile
+```
+
+This command would re-reconcile against the CredentialsRequests currently in the cluster. This is more of an on-going configuration management to ensure something doesn't change unexpectedly on us due to external factors. It would however support reducing permissions as a result of an additive only reconcile pre-upgrade.
+
+It should be noted that running this command after reconciling --available-updates, but before updating, would undo the pre-upgrade changes. As such this command should clear the reconciledAvailableUpdates list.
+
+
+## Open Questions [optional]
+
+ > 1. How can we handle net new CredentialsRequests in incoming release images?
+   * We could create the CredentialsRequest during the pre-upgrade reconciliation, but this does not guarantee that the target namespace for the component (where the Secret/ServiceAccount would be stored) exists.
+   * If we were to take the incoming CredentialsRequests, mind the credentials, and store in a temporary secret in the CCO namespace, we could later transfer it to the correct namespace when it exists.
+ > 1. Can the installer/CVO install manifests that contain status?
```

Comment:
yes, the installer can push manifests with status.

But the installer will race with CVO to create these CredentialRequest Objects with their status populated.

Reply:
Yeah this is my mistake, I thought release image contents were in manifests dir, that is not the case.